src/sentry/profiles/task.py

```diff
@@ -268,12 +259,12 @@ def _process_symbolicator_results_for_rust(profile: Profile, stacktraces: List[A
             frame.pop("context_line", None)
             frame.pop("post_context", None)
 
+        if len(symbolicated["frames"]) < 2:
+            continue
+
         # here we exclude the frames related to the profiler itself as we don't care to profile the profiler.
-        if (
-            len(symbolicated["frames"]) >= 2
-            and symbolicated["frames"][-1].get("function", "") == "perf_signal_handler"
-        ):
-            original["frames"] = symbolicated["frames"][:2]
+        if symbolicated["frames"][-1].get("function", "") == "perf_signal_handler":
+            original["frames"] = symbolicated["frames"][:-2]
```

Comment:
less important for the old format, but a comment here explaining it would be nice